### PR TITLE
Replace dangling else in forms table migration with explicit test on version 4

### LINKED ISSUE #1323: Replace dangling else in forms table migration with explicit test on version 4
We identified in #1252 a problem with the forms table migration. The dangling `else` in the `oldVersion` tests mean that the code in the migration to version 4 is run when downgrading.

The `else` should be replaced. I think what the `DatabaseHelper` in the `InstanceProvider` does with several disconnected `if` statements is pretty common. An alternative would be to introduce a switch statement.

### PR DESCRIPTION
**Pull request by grzesiek2010**
This PR is in reference to resolve #1323

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/provider/FormsProvider.java [hunk 1]
```diff
@@ -97,71 +98,101 @@ private void onCreateNamed(SQLiteDatabase db, String tableName) {
...
+            boolean success = true;
+            switch (oldVersion) {
+                case 1:
+                    success = upgradeToVersion2(db);
+                case 2:
+                case 3:
+                    success &= upgradeToVersion4(db, oldVersion);
+                    break;
+                default:
+                    Timber.i("Unknown version " + newVersion);
```
Comment (heyjamesknight) [inline collect_app/src/main/java/org/odk/collect/android/provider/FormsProvider.java]: I believe this should log 'oldVersion'.

File: collect_app/src/main/java/org/odk/collect/android/provider/FormsProvider.java [hunk 2]
```diff
@@ -180,53 +211,55 @@ public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
...
+                        + FormsProviderAPI.FormsColumns.JR_VERSION + ", "
+                        + FormsProviderAPI.FormsColumns.BASE64_RSA_PUBLIC_KEY + ", "
+                        + FormsProviderAPI.FormsColumns.JRCACHE_FILE_PATH + " FROM "
                         + TEMP_FORMS_TABLE_NAME);
                 db.execSQL("DROP TABLE IF EXISTS " + TEMP_FORMS_TABLE_NAME);
-
-                Timber.w("Successfully upgraded database from version %d to %d"
-                        + ", without destroying all the old data", oldVersion, newVersion);
+            } catch (SQLiteException e) {
+                Timber.i(e);
```
Comment (heyjamesknight) [inline collect_app/src/main/java/org/odk/collect/android/provider/FormsProvider.java]: Should this be Timber.e?

**Review by heyjamesknight — APPROVED** [review]
Two very minor logging suggestions. Otherwise looks good.

**Review by lognaturel — APPROVED** [review]

**Comment by lognaturel** [comment]
Thanks @grzesiek2010, @jknightco! I agreed with @jknightco's logging suggestions and made the changes.

**Pull request merged**